src/main.py

```diff
@@ -148,9 +150,16 @@ def analyze_line(line: str) -> str:
     leftovers_pat = re.compile("(.*) restored a little HP using its Leftovers!")
     black_sludge_pat = re.compile("(.*) restored a little HP using its Black Sludge!")
 
+
     fast_asleep_pat = re.compile("(.*) is fast asleep.")
     woke_up_pat = re.compile("(.*) woke up!")
 
+    damage_dealt_pat = re.compile("[0-9\.]*%")
+    super_effective_pat = re.compile("It's super effective!")
+    not_very_effective_pat = re.compile("It's not very effective...")
```

Comment:
If we're just matching strings, without trying to capture anything, then we don't need to compile extra regex patterns.